x/ccv/provider/keeper/throttle.go

```diff
@@ -322,16 +323,22 @@ func (k Keeper) QueueThrottledPacketData(
 	case ccvtypes.SlashPacketData:
 		bz, err = data.Marshal()
 		if err != nil {
+			// Slash packet data is assumed to be validated in OnRecvPacket and/or OnRecvSlashPacket,
+			// hence an error here would indicate something is very wrong.
 			panic(fmt.Sprintf("failed to marshal slash packet data: %v", err))
 		}
 		bz = append([]byte{slashPacketData}, bz...)
 	case ccvtypes.VSCMaturedPacketData:
 		bz, err = data.Marshal()
 		if err != nil {
+			// VSC matured packet data is assumed to be validated in OnRecvPacket and/or OnRecvVSCMaturedPacket,
```

Comment:
This assumption is wrong until #634 is solved btw, maybe that issue is worth prioritizing